cnd/src/btsieve/bitcoin/cache.rs

```diff
@@ -28,35 +24,55 @@ impl<C> Cache<C> {
     }
 }
 
+#[async_trait]
 impl<C> LatestBlock for Cache<C>
 where
     C: LatestBlock<Block = Block, BlockHash = BlockHash> + Clone,
 {
     type Block = Block;
     type BlockHash = BlockHash;
 
-    fn latest_block(
-        &mut self,
-    ) -> Box<dyn Future<Item = Self::Block, Error = anyhow::Error> + Send + 'static> {
+    async fn latest_block(&mut self) -> anyhow::Result<Self::Block> {
         let cache = Arc::clone(&self.block_cache);
         let mut connector = self.connector.clone();
 
-        let future = async move {
-            let block = connector.latest_block().compat().await?;
+        let block = connector.latest_block().await?;
 
-            let block_hash = block.bitcoin_hash();
-            let mut guard = cache.lock().await;
-            if !guard.contains(&block_hash) {
-                guard.put(block_hash, block.clone());
-            }
-
-            Ok(block)
+        let block_hash = block.bitcoin_hash();
+        let mut guard = cache.lock().await;
+        if !guard.contains(&block_hash) {
+            guard.put(block_hash, block.clone());
         }
-        .boxed()
-        .compat();
 
-        Box::new(future)
+        Ok(block)
     }
 }
 
-impl_block_by_hash!();
```

Comment:
Quick look on my phone, but it looks like you stopped using the macro yet you didn't delete it and actually updated it to be async. Am I wrong?

Reply:
Thanks @luckysori, you are right.  Deleted it now and force pushed.  (unrelated note: Are we aiming for correct discreet patches on PRs yet?)

Reply:
Aim for patches that yield a Rust binary :)